src/components/DependencyManager.tsx

```diff
@@ -167,73 +172,59 @@ const DependencyManager = ({ isOpen, onClose, onPipListUpdate = () => {} }) => {
     })();
   }, [depChanged]);
 
-  const runPipCommand = (args) => {
+  const runPipCommand = (args: string[]) => {
     setShellOutput('');
     setIsRunningShell(true);
-    const command = spawn(pythonKeys.python, ['-m', 'pip', ...args, '--disable-pip-version-check']);
+    // TODO: hope and pray pythonKeys is non-null
+    const command = spawn(pythonKeys!.python, [
+      '-m',
+      'pip',
+      ...args,
+      '--disable-pip-version-check',
+    ]);
 
     let outputString = '';
 
-    command.stdout.on('data', (data) => {
+    command.stdout.on('data', (data: unknown) => {
       outputString += String(data);
       setShellOutput(outputString);
     });
 
-    command.stderr.on('data', (data) => {
-      setShellOutput(data);
+    command.stderr.on('data', (data: unknown) => {
+      setShellOutput(String(data));
     });
 
     command.on('error', (error) => {
-      setShellOutput(error);
+      setShellOutput(String(error));
       setIsRunningShell(false);
     });
 
     command.on('close', (code) => {
-      console.log(`child process exited with code ${code}`);
+      console.log(`child process exited with code ${code ?? 'null'}`);
       setIsRunningShell(false);
     });
   };
 
-  const installPackage = async (dep) => {
-    setIsRunningShell(true);
-    setInstallingPackage(dep);
-    let output = '';
-    await pipInstallWithProgress(
-      pythonKeys.python,
-      dep,
-      (percentage) => {
-        setProgress(percentage);
-      },
-      (data) => {
-        output += String(data);
-        setShellOutput(output);
-      }
-    );
-    setIsRunningShell(false);
-    setProgress(0);
-  };
-
-  const updatePackage = async (dep) => {
+  const installPackage = async (dep: Dependency, upgrade: boolean) => {
     setIsRunningShell(true);
     setInstallingPackage(dep);
     let output = '';
     await pipInstallWithProgress(
-      pythonKeys.python,
+      // TODO: hope and pray pythonKeys is non-null
+      pythonKeys!.python,
       dep,
-      (percentage) => {
-        setProgress(percentage);
-      },
+      setProgress,
       (data) => {
-        output += String(data);
+        output += data;
```

Comment:
So idk if you tested this, and maybe I'm thinking of something else, but I'm pretty sure this is necessary or it just outputs a string of hex data

Reply:
I was wrong, it appears to work perfectly fine. I think I'm thinking of something in the file download/pip install file

Reply:
TS wouldn't allow the `+=` if `data` wasn't a string (or number). In this case, `pipInstallWithProgress` guarantees that `data` is a string.

Also, I think there are a few more unnecessary conversions. I think TS lint has a rule to detect those.